### ISSUE TITLE
Missing features for implementing basic `Result` methods

### ISSUE THREAD

**Issue by adlerjohn** [FuelLabs/sway#670]
The methods `is_ok()` and `is_err()` for `Result are currently implemented like this:

```rust
    /// Returns `true` if the result is [`Ok`].
    fn is_ok(self) -> bool {
        match self {
            Result::Ok(T) => {
                true
            },
            _ => {
                false
            },
        }
    }

    /// Returns `true` if the result is [`Err`].
    fn is_err(self) -> bool {
        match self {
            Result::Ok(T) => {
                false
            },
            _ => {
                true
            },
        }
    }
```

We would like the compiler to be able to handle an implementation like so:

```rust
    /// Returns `true` if the result is [`Ok`].
    fn is_ok(self) -> bool {
        match self {
            Ok(_) => true,
            _ => false,
        }
    }

    /// Returns `true` if the result is [`Err`].
    fn is_err(self) -> bool {
        !self.is_ok()
    }
```

**Comment by adlerjohn** [comment]
Oh, that original code is outdated and shouldn't work. The new code is here: https://github.com/FuelLabs/sway-lib-std/pull/68

**Comment by adlerjohn** [comment]
This was resolved with #815 and #913
